content/java/concepts/hash-map/hash-map.md

```diff
@@ -0,0 +1,217 @@
+    courseTeacher.put("Mathematics", "Jeanette");
+    courseTeacher.put("Physics", "Lily");
+
+    System.out.println(courseTeacher);
+    //Output: {Mathematics=Jeanette, History=Ben, Physics=Lily}
+  }
+}
+```
+
+## Accessing an Item
+
+In Java, an item can be accessed within a `HashMap` using the `.get()` method.
+
+The `.get()` method takes in one attribute which is the key and returns the corresponding value.
+
+```java
+courseTeacher.get("Physics");
+// Returns "Lily" which is the value for the key "Physics"
+```
+
+## Removing Items
+
+In Java, an item can be removed from a `HashMap` using the `.remove()` method.
+
+The `.remove()` method takes in 1 attribute which is the key and the method deletes the corresponding key-value pair from the `HashMap`.
+
+```java
+courseTeacher.remove("Physics");
+```
+
+```java
+import java.util.HashMap; // import the HashMap class
+
+public class Main {
+  public static void main(String[] args) {
+
+    HashMap<String, String> courseTeacher = new HashMap<String, String>();
+
+    // put(CourseName, Teacher)
+    courseTeacher.put("History", "Ben");
+    courseTeacher.put("Mathematics", "Jeanette");
+    courseTeacher.put("Physics", "Lily");
+
+    courseTeacher.remove("Physics");
+    System.out.println(courseTeacher);
+
+  }
+}
+
+// Output: { Mathematics = Jeanette, History = Ben }
+```
+
+The `clear` method can be used to delete all the items from the `HashMap`.
+
+```java
+courseTeacher.clear();
+```
+
+```java
+import java.util.HashMap; // import the HashMap class
+
+public class Main {
+  public static void main(String[] args) {
+
+    HashMap<String, String> courseTeacher = new HashMap<String, String>();
+
+    // put(CourseName, Teacher)
+    courseTeacher.put("History", "Ben");
+    courseTeacher.put("Mathematics", "Jeanette");
+    courseTeacher.put("Physics", "Lily");
+
+    courseTeacher.clear();
+    System.out.println(courseTeacher);
+
+  }
+}
+
+//Output: {}
+```
+
+## Traversing through a HashMap
+
+A `HashMap` can easily be traversed with the for-each loop.
+
+The `keySet()` method can be used to obtain only the keys while the `values()` method can be used to obtain only values.
+
+```codebyte/java
+import java.util.HashMap; // import the HashMap class
+
+public class Main {
+  public static void main(String[] args) {
+
+    HashMap<String, String> courseTeacher = new HashMap<String, String>();
+
+    courseTeacher.put("History", "Ben");
+    courseTeacher.put("Mathematics", "Jeanette");
+    courseTeacher.put("Physics", "Lily");
+
+    System.out.println("Courses offered at our Institute:");
+
+    for (String i : courseTeacher.keySet())
+      System.out.println(i);
+
+    System.out.println("\nTeachers teaching at our Institute:");
+
+    for (String i : courseTeacher.values())
+      System.out.println(i);
+
+  }
+}
+
+/*
+Output:
+Courses offered at our Institute:
+Mathematics
+History
+Physics
+
+Teachers teaching at our Institute:
+Jeanette
+Ben
+Lily
+*/
```

Comment:
Since this output is pretty big, let's move this multi-line comment out and rewrite as something like this: 

The output will be:

```
Courses offered at our Institute:
Mathematics
History
Physics
Teachers teaching at our Institute:
Jeanette
Ben
Lily
```